enum class DAY {
    SUNDAY, MONDAY, TUESDAY, WEDNESDAY, THURSDAY, FRIDAY, SATURDAY
}

fun foo(x: DAY) = when (x) {
    DAY.MONDAY, DAY.TUESDAY, DAY.WEDNESDAY, DAY.THURSDAY, DAY.FRIDAY -> <!UNRESOLVED_REFERENCE!>print<!>("Weekday")
    DAY.SATURDAY, DAY.SUNDAY -> <!UNRESOLVED_REFERENCE!>print<!>("Weekend")
}
